### PR TITLE
Feature/classes testing

### PR DESCRIPTION
**Pull request by Matt-Roden**
- [x] Feature
- [ ] Fix

## What was done?

Wrote all class tests and implementation code. All tests are passing

## How can this be tested?

Pull down repo and from the root directory in repo, run `npm install` and then `npm test` 

## Where should the reviewer start?

In the src directory, looking at each class as well as in the test folder, looking at each test

### CODE REVIEW

File: src/Traveler.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    this.name = traveler.name;
+    this.type = traveler.travelerType;
+    this.trips = tripsData.filter((trip) => trip.travelerID=== this.id); //will be a filtered array of all trip objects that have the same userID as this traveler and the date is prior to today's date
+    // this.upcomingTrips = tripsData.filter((trip) => trip.id === this.id && trip.status === 'pending');// will be filtered array of all trip objects that have the same userID as this traveler and the date is later than today's date
+    // this.currentTrips = [] // will be filtered array of all trip objects that have the same userID as this traveler and the date is the same as today's date
+    // this.pendingTrips = [] // will be filtered array of all trip objects that have the same userID as this traveler and the status is pending
+  }
+  calculateTotalSpentThisYear(year) {
+    console.log(this.trips)
+    const nonPenidngTrips = this.trips.filter((trip) => trip.status !== 'pending')
```
Comment (ericli1996) [inline src/Traveler.js:13]: nonPendingTrips typo

File: src/Traveler.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Traveler {
+  constructor(traveler, tripsData) {
```
Comment (ericli1996) [inline src/Traveler.js:2]: I like how you included tripsData as a parameter for this class.

File: src/Trip.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(trip, destinations) {
+    this.id = trip.id;
+    this.travelerID = trip.userID;
+    this.destinationID = trip.destinationID;
+    this.travelersAmount = trip.travelers;
+    this.startDate = trip.date;
+    this.duration = trip.duration;
+    this.status = trip.status || 'pending';
+    this.suggestedActivities = trip.suggestedActivities;
+    this.destinationName = destinations.findDestination(trip.destinationID).destination;
```
Comment (ericli1996) [inline src/Trip.js:11]: nice method for importing the Destination data into the trip class!

File: src/Traveler.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    this.id = traveler.id;
+    this.name = traveler.name;
+    this.type = traveler.travelerType;
+    this.trips = tripsData.filter((trip) => trip.travelerID=== this.id); //will be a filtered array of all trip objects that have the same userID as this traveler and the date is prior to today's date
+    // this.upcomingTrips = tripsData.filter((trip) => trip.id === this.id && trip.status === 'pending');// will be filtered array of all trip objects that have the same userID as this traveler and the date is later than today's date
+    // this.currentTrips = [] // will be filtered array of all trip objects that have the same userID as this traveler and the date is the same as today's date
+    // this.pendingTrips = [] // will be filtered array of all trip objects that have the same userID as this traveler and the status is pending
+  }
+  calculateTotalSpentThisYear(year) {
```
Comment (ericli1996) [inline src/Traveler.js:11]: Great job calculating total spent based only on approved trips.

**Review by ericli1996 — APPROVED** [review]
Overall looks very good. You may have a typo here and there in variable naming (check notes) so resolve those. Testing looks solid but needs sad paths to go that extra mile. Great work!

File: src/Traveler.js (hunk 1 above)
Comment (Matt-Roden) [reply src/Traveler.js:13]: Good catch, thanks man!

**Pull request merged**